src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -591,7 +591,14 @@ impl StreamSink {
         if let Some(sink_type) = properties.get(SINK_TYPE_OPTION) {
             let sink_type = match sink_type.as_str() {
                 SINK_TYPE_APPEND_ONLY => SinkType::AppendOnly,
-                SINK_TYPE_UPSERT => SinkType::Upsert,
+                SINK_TYPE_UPSERT => {
```

Comment:
Does it mean that we want to keep current syntax of using `sink_type = 'upsert'` to represent non-append-only, instead of using `retract` which seems more correct in semantics?

Reply:
+1

Reply:
Using `retract` is more accurate, but `upsert` has been used in our docs and users, if we change this behavior suddenly, I think it will confuse our users. So I think we need to update our doc about iceberg sink to recommend users to use  `retract` instead of `upsert`, but keep `upsert` semantics as `retract` for iceberg (for a time).